Add custom toc file to exclude page title from docs nav

cfarm opened the pull request:
Fixes annoying issue where the h1/page title from documentation pages gets included in the sidebar navigation alongside the rest of the child headings. Now, only h2s and h3s will appear in the sidebar nav.

## Additions

- `docs-theme-overrides` folder for [customizing](https://www.mkdocs.org/user-guide/styling-your-docs/#using-the-theme-custom_dir) the Read the Docs theme.  note that this folder has to live in the same directory as the `docs` folder, it cannot be nested inside it, sadly.
- custom `toc.html` template file based on latest 1.0.4 of mkdocs release, with edit noted below

## Removals

- This line that includes the page title in the TOC output: https://github.com/mkdocs/mkdocs/blob/1.0.4/mkdocs/themes/readthedocs/toc.html#L2 the toc template is only included in the nav, and not in other parts of the theme, so no other functionality is affected.

## Changes

-

## Testing

1.

## Screenshots



| Before...eww  | After..aaaa |
|---|---|
| ![screen shot 2019-01-08 at 5 49 46 pm](https://user-images.githubusercontent.com/702526/50863785-d0fb3d80-136d-11e9-9860-14a330093374.png)  |  ![screen shot 2019-01-08 at 5 50 06 pm](https://user-images.githubusercontent.com/702526/50863798-dbb5d280-136d-11e9-8203-689d656dd108.png) |
|  ![screen shot 2019-01-08 at 5 50 29 pm](https://user-images.githubusercontent.com/702526/50863812-ecfedf00-136d-11e9-8243-0fd2fc0c3e68.png)  | ![screen shot 2019-01-08 at 5 50 37 pm](https://user-images.githubusercontent.com/702526/50863811-ea9c8500-136d-11e9-88f0-0244f7614f48.png)  |





## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist

### Browsers

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Internet Explorer 8, 9, 10, and 11
- [ ] Edge
- [ ] iOS Safari
- [ ] Chrome for Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

chosak commented on mkdocs.yml:
```diff
@@ -32,7 +32,8 @@ nav:
 theme:
-  name: 'readthedocs'
+    name: readthedocs
+    custom_dir: docs-theme-overrides/
```
Per [the docs](https://www.mkdocs.org/user-guide/configuration/#custom_dir), this can be a directory relative to the MkDocs configuration file. If I move the overrides dir to `docs/theme-overrides` and change this here to the same (either `docs/theme-overrides` or `./docs/theme-overrides`), it works for me locally (`mkdocs serve`) without cluttering up the repository root.

I see your comment that it cannot be nested; is there some other way of serving and building that doesn't work with a relative directory?

cfarm replied:
doh, i thought i tried it but i had forgotten to update the file path in the config here. thanks, i'll push this change.

willbarton commented:
Did we confirm that the MkDocs master fixes the problem?

cfarm commented:
I don't know how to test the whole theme from the master branch, but trying to just include the toc.html as a local override to see if the updates to that fix the issue doesn't work, because there are template tags referenced that don't match the rest of the theme code from v1.0.4

https://github.com/mkdocs/mkdocs/blob/master/mkdocs/themes/readthedocs/toc.html

cfarm commented:
I moved the theme-overrides into the docs folder as suggested, and added a TODO note about possibly removing it if the next Mkdocs release fixes this issue.
